Panic in ubbagent

A customer marketplace deployment hit the following issue in ubbagent:

```
panic: RetryingSender.maybeSend: loading from retry queue: unexpected end of JSON input
goroutine 15 [running]:
vendor/github.com/GoogleCloudPlatform/ubbagent/pipeline/senders.(*RetryingSender).maybeSend(0xc0005806e0, 0xbf97a99c50737bd1, 0x24e27478, 0x48378c0)
	/vendor/github.com/GoogleCloudPlatform/ubbagent/pipeline/senders/retry.go:199 +0x7d8
vendor/github.com/GoogleCloudPlatform/ubbagent/pipeline/senders.(*RetryingSender).run(0xc0005806e0, 0xbf97a99c50737bd1, 0x24e27478, 0x48378c0)
	/vendor/github.com/GoogleCloudPlatform/ubbagent/pipeline/senders/retry.go:149 +0x6a
created by vendor/github.com/GoogleCloudPlatform/ubbagent/pipeline/senders.newRetryingSender
	/vendor/github.com/GoogleCloudPlatform/ubbagent/pipeline/senders/retry.go:88 +0x244
```

--- a/persistence/disk.go
+++ b/persistence/disk.go
@@ -61,6 +61,9 @@ func (v *diskValue) load(obj interface{}) error {
 	if err != nil {
 		return err
 	}
+	if len(jsontext) == 0 {
+		return ErrNotFound
+	}
 	err = json.Unmarshal(jsontext, obj)
 	if err != nil {
 		return err

--- a/persistence/disk_test.go
+++ b/persistence/disk_test.go
@@ -1,0 +1,43 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package persistence
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+)
+
+type emptyStruct struct{}
+
+func TestEmptyFile(t *testing.T) {
+	tmpdir, err := ioutil.TempDir("", "disk_endpoint_test")
+	if err != nil {
+		t.Fatalf("Unable to create temp directory: %+v", err)
+	}
+	defer os.RemoveAll(tmpdir)
+
+	ioutil.WriteFile(path.Join(tmpdir, "empty.json"), []byte{}, 0644)
+	p, err := NewDiskPersistence(tmpdir)
+	if err != nil {
+		t.Fatalf("Unable to create new DiskPersistence")
+	}
+	v := emptyStruct{}
+	err = p.Value("empty").Load(&v)
+	if err != ErrNotFound {
+		t.Fatalf("Expected NotFound error but found %+v", err)
+	}
+}

